Content.Client/Audio/AmbientSoundSystem.cs

```diff
@@ -0,0 +1,178 @@
+using System.Collections.Generic;
+using System.Linq;
+using Content.Shared.Audio;
+using Content.Shared.CCVar;
+using Robust.Client.Player;
+using Robust.Shared.Audio;
+using Robust.Shared.Configuration;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+
+namespace Content.Client.Audio
+{
+    /// <summary>
+    /// Samples nearby <see cref="AmbientSoundComponent"/> and plays audio.
+    /// </summary>
+    public sealed class AmbientSoundSystem : SharedAmbientSoundSystem
+    {
+        [Dependency] private IEntityLookup _lookup = default!;
+        [Dependency] private readonly IPlayerManager _playerManager = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+
+        private int _maxAmbientCount;
+
+        private float _maxAmbientRange;
+        private float _cooldown;
+        private float _accumulator;
+
+        /// <summary>
+        /// How many times we can be playing 1 particular sound at once.
+        /// </summary>
+        private int _maxSingleSound = 3;
+
+        private Dictionary<AmbientSoundComponent, (IPlayingAudioStream? Stream, string Sound)> _playingSounds = new();
+
+        private const float RangeBuffer = 0.1f;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            var configManager = IoCManager.Resolve<IConfigurationManager>();
```

Comment:
cache?

Reply:
It's only using it on startup and shutdown so not needed IMO given it does the C# event subscribe. It's 1 reference for 1 system so not a huge deal on whether it's cached or not.